src/main/java/com/onlineshop/be/services/ProductServiceImplementation.java

```diff
@@ -0,0 +1,122 @@
+package com.onlineshop.be.services;
+
+import com.onlineshop.be.enums.OrderAndProductStatus;
+import com.onlineshop.be.enums.Results;
+import com.onlineshop.be.exceptions.CustomizedExceptions;
+import com.onlineshop.be.model.ProductInfo;
+import com.onlineshop.be.repository.ProductInfoRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+@Service
+public class ProductServiceImplementation implements ProductService {
+
+    @Autowired
+    ProductInfoRepo productInfoRepository;
+
+    @Autowired
+    CategoryService categoryService;
+
+    @Override
+    public ProductInfo findOne(String productId) {
+        return productInfoRepository.findById(productId).get();
+    }
+
+    @Override
+    public Page<ProductInfo> findUpAll(Pageable pageable) {
+        return productInfoRepository.findAllByCategoryTypeOrderByIdAsc(OrderAndProductStatus.UP.getCode(),pageable);
+    }
+
+    @Override
+    public Page<ProductInfo> findAll(Pageable pageable) {
+        return productInfoRepository.findAllByOrderById(pageable);
+    }
+
+    @Override
+    public Page<ProductInfo> findAllInCategory(Integer categoryType, Pageable pageable) {
+        return productInfoRepository.findAllByCategoryTypeOrderByIdAsc(categoryType, pageable);
+    }
+
+    @Override
+    @Transactional
+    public void increaseStock(String productId, int amount) {
+        ProductInfo productInfo = findOne(productId);
+        if (productInfo == null) throw new CustomizedExceptions(Results.PRODUCT_NOT_EXIST);
```

Comment:
after if condition it is recomended to have {} even though it is only 1 line of code